pennylane/ops/op_math/adjoint.py

```diff
@@ -21,13 +21,16 @@
 from pennylane.operation import Observable, Operation, Operator
 from pennylane.queuing import QueuingManager
 from pennylane.tape import make_qscript
+from pennylane.compiler import compiler
+from pennylane.compiler.compiler import CompileError
 
 from .symbolicop import SymbolicOp
 
 
 # pylint: disable=no-member
 def adjoint(fn, lazy=True):
-    """Create the adjoint of an Operator or a function that applies the adjoint of the provided function.
+    """A :func:`~.qjit` compatible adjoint transformation that creates the adjoint of an Operator or a function
+    that applies the adjoint of the provided function.
 
     Args:
         fn (function or :class:`~.operation.Operator`): A single operator or a quantum function that
```

Comment:
@rmoyard I can't comment on it, but for the `lazy` kwarg, I would add the follwing:

```rest
lazy=True (bool): If the transform is behaving lazily, all operations are
    wrapped in a ``Adjoint`` class and handled later. If ``lazy=False``,
    operation-specific adjoint decompositions are first attempted.
    This argument is not supported when used with :func:`~.qjit`.

Reply:
Thanks I will add that !